tools/scxtop/src/app.rs

```diff
@@ -2073,10 +2073,10 @@ impl<'a> App<'a> {
             Action::PageDown => self.on_pg_down(),
             Action::Enter => self.on_enter(),
             Action::SetState(state) => {
-                if state == self.state {
+                if *state == self.state {
                     self.set_state(self.prev_state.clone());
                 } else {
-                    self.set_state(state);
+                    self.set_state(state.clone());
```

Comment:
nit: if you derive `Copy` on `AppState` this becomes a `*`, which is good because it shows we don't need to clean it up in future.